### PR TITLE
Implement basic rand()

### LINKED ISSUE #6997: Stdlib rand() in global ctor causes hard fault
#### Problem

The PR for #5389 introduced a new global object for the unencrypted message counter. This c++ object is initialized at global scope with a pseudorandom number from `rand()`. In this case, `rand()` is supplied by the `newlib-nano` standard library. This implementation of `rand()` is meant to be reentrant, so it calls the FreeRTOS function to get the reent struct. Which ends up attempting to alloc memory for the new reent struct. The system heap is not setup on the TI platform until `main`, causing a hard fault.

 #### Proposed Solution

Eclipsing the stdlib implementation of `rand()` and `srand()` is the simplest change. A basic non-reentrant function will work for the PRNG assumption here. The TRNG and DRBG can be fulfilled by other sources where needed.

### PR DESCRIPTION
**Pull request by srickardti**
Disabled shadowing of the FreeRTOS config by the DMM header.
Implemented a basic PRNG for ctor startup.

The global initialization of the global message counter struct in
`src/transport/MessageCounter.h` results in a call to the C standard library
function `rand()` from a C++ ctor. This results in a call to pvPortMalloc,
causing a hard fault with the current buffer implementation.

 #### Problem

`rand()` is called before `main()` so the calls to allocate a reent struct fail.

 #### Summary of Changes

Added a basic non-reent PRNG for the TI platform.

Fixes #6997

#### Testing

Manually tested with an LP_CC2652R7. Execution was traced from the `ResetISR` to `main` with CCS and the device was left to free run. log output was viewed on the debug UART connection and the processor was halted in the power idle loop.

### CODE REVIEW

**Review by woody-apple — CHANGES_REQUESTED** [review]
Thanks for the change! We have updated our PR template, requesting changes to conform to the PR template:
```
#### Problem
What is being fixed?  Examples:
* Fix crash on startup
* Fixes #12345 Frobnozzle is leaky (exactly like that, so GitHub will auto-close the issue).

#### Change overview
What's in this PR

#### Testing
How was this tested? (at least one bullet point required)
* If unit tests were added, how do they cover this issue?
* If unit tests existed, how were they fixed/modified to prevent this in future?
* If new unit tests are not added, why not?
* If integration tests were added, how do they verify this change?
* If new integration tests are not added, why not?
* If manually tested, what platforms controller and device platforms were manually tested, and how?
* If no testing is required, why not?
```

**Comment by srickardti** [comment]
> 
> 
> Thanks for the change! We have updated our PR template, requesting changes to conform to the PR template:
>

Apologies, I must have gotten it in before the template was update. I've included the testing information now.

**Review by woody-apple — APPROVED** [review]
No apologies needed, the template is new :) Thanks!

**Review by LuDuda — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

File: src/platform/cc13x2_26x2/Random.c [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <stdlib.h>
+
+/* Basic implementation of psudo-random number generation to override the
```
Comment (bzbarsky-apple) [inline src/platform/cc13x2_26x2/Random.c:21]:
```suggestion
/* Basic implementation of pseudo-random number generation to override the
```

**Pull request merged**